### ISSUE TITLE
[FEATURE] Langchain4j PGvector - Add option for mutiple metadata

### ISSUE THREAD

**Issue by shyamramath** [langchain4j/langchain4j#1926]
**Is your feature request related to a problem? Please describe.**

I'm looking for a feature to insert a document type of data to PG vector using langchain4j .  Add more metadata in to the embeddings .

I was testing the example code below, there only metadata can be added. 
https://github.com/langchain4j/langchain4j-examples/blob/main/pgvector-example/src/main/java/PgVectorEmbeddingStoreExample.java

**Describe the solution you'd like**

I was reading the Langchain document found the below article, can we some option like the document in the below article 
The below code snippet has lot of metadata. Can we have something similar here as well in Langchain4j
Document(
        page_content="there are cats in the pond",
        metadata={"id": 1, "location": "pond", "topic": "animals"},
    )

https://python.langchain.com/docs/integrations/vectorstores/pgvector/

**Describe alternatives you've considered**

**Additional context**

**Comment by langchain4j** [comment]
@shyamramath do you want to store metadata along with the document text?

You can do so:

```java
Metadata metadata = Metadata.from(Map.of(
    "id", 1,
    "topic", "sport"
));
TextSegment textSegment = TextSegment.from("I like football.", metadata);
```

**Comment by langchain4j** [comment]
See more details here: https://docs.langchain4j.dev/tutorials/rag#metadata

**Comment by shyamramath** [comment]
Thanks for sharing the info . 

The below snippet also does the same as you mentioned in your comment ? or is there any difference passing metadata as a Map. 

Map<String,String> mymetadata = new HashMap<>();
            mymetadata.put("id", "1");
            mymetadata.put("topic", "sport");
            mymetadata.put("dept", "English");
            TextSegment segment3 = TextSegment.from("I like football", new Metadata(mymetadata));

**Comment by langchain4j** [comment]
@shyamramath yes, it does the same. In both cases `Metadata` is created from a `Map`
